### DISCUSSION TITLE
Unable to choose correct device type for ElectriQ EcoSilent 14HPW

### DISCUSSION THREAD

**Discussion by tapp1993** [make-all/tuya-local#1130] [Q&A]
I'm trying to add my ElectriQ EcoSilent 14HPW and can see that it's a supported device type and there's a YAML file for it.

However, when I add in the device details and get to the stage where I pick the device type from the dropdown list, the correct device type doesn't come up, and instead it suggests the Daewoo Dhome OL-A011.

Is there any way I can override this?

**Comment by JackJackattack4** [comment]
I have the same issue (the device I want to add is there but does not appear in the drop down menu). The work around I used was to "hi-jack" the yaml file of one on the devices from the drop down menu by replacing its content with the yaml of the device I want.

Any other solutions out there?

**Comment by awensaunders** [comment]
I'm running into the same issue with the EcoSilent 12WAP, which is a related model. I can only select a dozen or so device types, none of which are the correct types.

<img width="717" alt="Screenshot 2024-04-08 at 16 43 55" src="https://github.com/make-all/tuya-local/assets/3837160/3db2d922-0e26-436c-a3e3-7f9234f77365">

**Reply by make-all** [reply]
If you look at the log, there should be a message output with the dps that were matched against the device configs. With this info, it might be possible to improve the config to match your device, but without it, there is nothing to recommend.  Hacky solutions of manually changing config files are not to be recommended, as they may come with undesirable side effects.

**Reply by awensaunders** [reply]
Here are the dps that were emitted by the logs: 

```
Device matches daewoo_dhome_heatpump with quality of 78%. DPS: {"updated_at": 1712616485.1183345, "1": false, "2": 18, "3": 18, "4": "cool", "5": "f5", "19": "c", "22": 0, "101": true, "102": false}
```

I have a working config yaml file on my fork, largely based on the ecosilent14hpw, but the auto detection doesn't work for it. I've attached the yaml here for your convenience. My guess is the call to `device.async_possible_types()` is not returning the config I want as a possible config, but I haven't dug too much into it.

I'll happily open a PR with the config file for the electriq_ecosilent123_wap_aircon once this is resolved.

[electriq_ecosilent12wap_aircon.yaml.txt](https://github.com/make-all/tuya-local/files/14911602/electriq_ecosilent12wap_aircon.yaml.txt)

**Comment by make-all** [comment]
It looks like just the fahrenheit dps are missing from your log, so need to be marked optional in the config.
